File: HostbooksWeb-app/src/main/java/com/hostbooks/app/Service/EmployeeService.java
package com.hostbooks.app.Service;

import com.hostbooks.app.Exception.EmployeeException;
import com.hostbooks.app.Model.Employee;

import java.util.List;

public interface EmployeeService {

    public Employee addNewEmployee(Employee employee);

    public List<Employee> getAllEmployees() throws EmployeeException;

    public Employee getEmployeeById(Integer employeeId) throws EmployeeException;

    public Employee updateEmployee(Employee employee)throws EmployeeException;

    public Employee deleteEmployee(Integer employeeId)throws EmployeeException;

}

File: hostbooks-webapp-angular/src/app/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Employee } from './employee';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseURL = "http://localhost:8081/hostbooks-app/v1/employee";

  constructor(private httpClient: HttpClient) { }
  
  getEmployeeList(): Observable<Employee[]>{
    return this.httpClient.get<Employee[]>(`${this.baseURL}/all`);
  }

  addEmployee(employee:Employee):Observable<Employee[]>{
    return this.httpClient.post<Employee[]>(`${this.baseURL}/addEmployee`,employee);
  }

  updateEmployee(employee:Employee):Observable<Employee[]>{
    return this.httpClient.post<Employee[]>(`${this.baseURL}/updateEmployee`,employee);
  }

  getEmployee(employeeId:number):Observable<Employee>{
    return this.httpClient.get<Employee>(`${this.baseURL}/getEmployee/${employeeId}`);
  }

  deleteEmployee(employeeId:number):Observable<Employee>{
    return this.httpClient.delete<Employee>(`${this.baseURL}/deleteEmployee/${employeeId}`);
  }
}
